metadata-ingestion/src/datahub/ingestion/source/aws/sagemaker_processors/models.py

```diff
@@ -385,6 +388,27 @@ def strip_quotes(string: str) -> str:
             model_metrics,
         )
 
+    @staticmethod
+    def get_group_name_from_arn(arn: str) -> str:
+        """
+        Extract model package group name from a SageMaker ARN.
+
+        Args:
+            arn (str): Full ARN of the model package group
+
+        Returns:
+            str: Name of the model package group
+
+        Example:
+            >>> arn = "arn:aws:sagemaker:eu-west-1:123456789:model-package-group/my-model-group"
+            >>> get_group_name_from_arn(arn)
+            "my-model-group"
```

Comment:
you can use doctest to automatically turn this into a unit test

we have a few examples of this in our codebase already